README.md

```diff
@@ -8,20 +8,432 @@
 
 ## Overview
 
-We want to continue to give students a ton of practice with state changes in
-their components. In many ways, this lesson repeats stuff that (at least as the
-course is currently laid out) students have already seen. The main focus is on
-adding complexity — more nesting, more complicated components (dropdowns,
-popovers, etc.) — and giving students a _ton_ of practice.
+React allows us to structure applications using small, reusable components, but
+React isn't just about creating static pages, it's about building **react**ive
+user interfaces that enable us to coordinate complex user interactions.
 
-In this lesson, we could have students build a simple popover component (like
-what Medium uses for logging in). It's fairly basic, but it should be easy to
-trigger from a bunch of different places (but not as easy as, say, using Flux —
-but that comes later).
+In this lesson we will be learning about how we can update the state of
+components.
 
-The `Popover` component can take an `on` prop from its parent component, and the
-parent is free to control that prop however it wants.
+## State vs Props
+
+Managing state is hard. React allows you to encapsulate two different "kinds" of
+state: "internal" state that can only be mutated and accessed by the component
+that manages it (`state`) and date that gets passed down to the component that
+renders it (`props`). This sounds way more complicated than it actually is, so
+let's take a step back and look at a simple example: A blog. The following
+examples are going to concentrate on a `Blog` component that renders multiple
+`BlogPost`s. Visually, this might look something like this:
+
+![Blog](./diagrams/overview.png)
+
+### Pure Components
```

Comment:
I think we can just remove this entire section, since this is more general/basic state vs props stuff that gets taught earlier on in the course